providers/azure.go

```diff
@@ -126,6 +125,15 @@ func (p *AzureProvider) Redeem(ctx context.Context, redirectURL, code string) (*
 	if p.ProtectedResource != nil && p.ProtectedResource.String() != "" {
 		params.Add("resource", p.ProtectedResource.String())
 	}
+	return params, nil
+}
+
+// Redeem exchanges the OAuth2 authentication token for an ID token
+func (p *AzureProvider) Redeem(ctx context.Context, redirectURL, code string) (*sessions.SessionState, error) {
+	params, err := p.prepareRedeem(redirectURL, code)
```

Comment:
If we did this: https://github.com/oauth2-proxy/oauth2-proxy/pull/1021#discussion_r571505330

Could we move Redeem to use the `oauth2.Config` like this: https://github.com/oauth2-proxy/oauth2-proxy/blob/d2ffef2c7e0a622295c42cb2fdd70ef14593d03c/providers/oidc.go#L36

And not need to manually make our own `token_endpoint` call from scratch in the helper method?

Reply:
This is me just thinking out loud -- ignore this for this PR; I think the Azure `resource` parameter needs more thorough investigation outside of this PR.